src/amount.h

```diff
@@ -15,4 +15,16 @@ typedef int64_t CAmount;
 static const CAmount COIN = 100000000;
 static const CAmount CENT = 1000000;
 
+/** No amount larger than this (in satoshi) is valid.
+ *
+ * Note that this constant is *not* the total money supply, which in Bitcoin
+ * currently happens to be less than 21,000,000 PIV for various reasons, but
+ * rather a sanity check. As this sanity check is used by consensus-critical
+ * validation code, the exact value of the MAX_MONEY constant is consensus
+ * critical; in unusual circumstances like a overflow bug that allowed
+ * for the creation of coins out of thin air modification could lead to a fork.
+ * */
+static const CAmount MAX_MONEY = 21000000 * COIN;
```

Comment:
if it's all the same, i'd rather name this `MAX_MONEY_OUT` or `MAX_MONEY_TX`.

`MAX_MONEY` by itself is not sufficient to self-explain what it is, and has led to confusion in the past, and was one of the reasons to use `nMaxMoneyOut` in the chain params. Even upstream, many agreed that the variable name was confusing, but because the end-all total supply of Bitcoin will end up being LESS than this value, they have decided it is okay to leave it as-is.